### ISSUE TITLE
Update Checkstyle To Match Requirements of CONTRIBUTING.md

### ISSUE THREAD

**Issue by donnfelker** [realm/realm-java#2016]
Would be great if the checkstyle was updated to match the recommendations/requirements that are present in https://github.com/realm/realm-java/blob/master/CONTRIBUTING.md

**Comment by cmelchior** [comment]
Yes, as much as possible should really be automatically checked.
